### ISSUE TITLE
Questions on no_information_rate calculation

### ISSUE THREAD

**Issue by TimZaragori** [rasbt/mlxtend#687]
I am trying to modify your `no_information_rate` function to make it suitable for `roc_auc_score` function of `sklearn` for example which takes as argument the output of `estimator.decision_function(X)` for binary problems. Thus, it doesn't work in the current form of the function where you get the output of `estimator.predict(X)` as argument. I have some idea to generalize the calculation for any scorer but it doesn't work with the current implementation and especially with `np.meshgrid` so I tried to use something else but I can't find the same results as your function. 
As far as I understood the equation given in the original publication, here is what I should do in case of accuracy:
```
import numpy as np
from itertools import product
from sklearn.metrics import accuracy_score

for i in range(y.shape[0]):
    for j in range(y.shape[0]):
        pred = est.predict(X[j, :].reshape(1, -1))
        combinations.append(1 if pred == y[i] else 0)
no_information_rate = np.mean(combinations)

# Or equivalent to
combinations = []
for i in range(y.shape[0]):
    for j in range(y.shape[0]):
        combinations.append(np.array([y[i], est.predict(X[j, :].reshape(1, -1))[0]]))
no_information_rate = accuracy_score(np.array(combinations)[:, 0], np.array(combinations)[:, 1])

# Or more simplier
combinations = np.array(list(product(y, est.predict(X))))
no_information_rate = accuracy_score(combinations[:, 0], combinations[:, 1])
```

Am I misunderstanding something in the calculation of the no information rate?

**Comment by rasbt** [comment]
The general definition of the no information rate is the average loss of all possible combinations of class y_i 's and x_i's, where y's are the class labels, and x's are the features for i=1...n (n is the size of the dataset).

I think your 3 methods look all correct to me, but they are maybe a bit expensive. For example, in 

```python
for i in range(y.shape[0]):
    for j in range(y.shape[0]):
        pred = est.predict(X[j, :].reshape(1, -1))
        combinations.append(1 if pred == y[i] else 0)
no_information_rate = np.mean(combinations) 
```

you are computing `est.predict(X[j, :].reshape(1, -1)` i times, but you only need to compute it once. 

In any case, I am just looking at how I implemented it, and I think what I was going for was having 

    combinations = np.array(list(product(y, est.predict(X))))

implemented more efficiently by using 

    combinations = np.array(np.meshgrid(targets, predictions)).reshape(-1, 2)

For some reason, the meshgrid function, does something different. I think the current implementation needs to be fixed to 

    combinations = np.array(list(product(targets, predictions)))

**Comment by TimZaragori** [comment]
Thanks for your answer and your work. I might have found a typo just down these lines on the relative overfitting rate calculation.
Current line is : `(-(test_acc - train_acc)) / (gamma - (1 - test_acc))`
Shouldn't this be : `(-(test_acc - train_acc)) / (gamma - (1 - train_acc))` or maybe `(-(test_acc - train_acc)) / (gamma - (- train_acc))` since there isn't the `1-x` in the numerator?

**Comment by rasbt** [comment]
Thanks again, I checked with the original paper, and it should have been "train acc" indeed:
![Screen Shot 2020-05-11 at 10 29 40 AM](https://user-images.githubusercontent.com/5618407/81581714-b75c7280-9374-11ea-8dcd-df9c0927fac3.png)
(where \bar{err} is the resubstitution or training error)

It's been a long time ago, but I remember getting funny results with that initially. But maybe this was due to something in the no-information-rate calculation. In any case, now making the two changes outlined here, the outputs are almost the same as before (see unit tests in #688; 0.96528 vs 0.9649 ie rounded;  0.65034 vs 0.64831, ie rounded 0.650) but hopefully calculated correctly now :).
